### ISSUE TITLE
Qt backend figureoptions.py does not work due to change in image.py

### ISSUE THREAD

**Issue by bastianbeischer** [matplotlib/matplotlib#9332]
### Bug report

**Bug summary**

Clicking on the figure options button (only available in the Qt backend) when showing an image does not work anymore. The dialog does not open and the script throws a TypeError.

**Code for reproduction**

```python
from matplotlib import use
use('Qt5Agg')

import numpy as np
import matplotlib.pyplot as plt

data = np.ones((10,10))

plt.figure()
plt.imshow(data)
plt.show()
```

Click on figure options button (the second one from the right).

**Actual outcome**

Figure options dialog is not shown, instead a TypeError is thrown:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 681, in edit_parameters
    figureoptions.figure_edit(axes, self)
  File "/usr/lib/python2.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py", line 168, in figure_edit
    + [(name, name) for name in sorted(image.iterpnames)])]
TypeError: 'instancemethod' object is not iterable
```

**Expected outcome**

Figure options dialog is opened and the script does not throw the TypeError. Used to work in matplot 2.0.2.

**Matplotlib version**

Bug introduced in matplotlib 2.1.0, specifcally this commit:

https://github.com/matplotlib/matplotlib/commit/6ee70cb49705f9057c5716f9bc2a4147eead5442

changes 'iterpnames' from a variable (which is iterable) into a function (which must be called). The callside (figureoptions.py) was not modified accordingly. I guess adding '()' after iterpnames in figureoptions.py line 168 fixes the problem (untested).

**Comment by anntzer** [comment]
Sorry... xref #9335.

**Comment by tacaswell** [comment]
Closed by #9335 

Thanks for taking the time to report this @bastianbeischer !
